### PR TITLE
ARROW-3866: [Python] Column metadata is not transferred to tables in pyarrow

### PR DESCRIPTION
**Pull request by tanyaschlusser**
Use columns' existing metadata to create the new fields in `Table.from_arrays()`.
Also persists the original `nullable` value.

Happy to change things! Thank you for putting a newbie label on it.

### CODE REVIEW

**Comment by kszucs** [comment]
@tanyaschlusser Thank You for the PR! :)

Would You be interested in resolving similar issues?

File: python/pyarrow/tests/test_table.py [hunk 1]
```diff
@@ -578,6 +578,14 @@ def test_table_basics():
 
     assert table.columns == columns
 
+    # Added to test https://issues.apache.org/jira/browse/ARROW-3866
+    columns = []
+    for col in table.itercolumns():
+        new_field = col.field.add_metadata({"a": "A", "b": "B"})
+        columns.append(pa.column(new_field, col.data))
+    table = pa.Table.from_arrays(columns)
+    assert b"a" in table.column(0).field.metadata
```
Comment (wesm) [inline python/pyarrow/tests/test_table.py]: Do you want to add a test to show that you can have fields with `nullable=False`?

**Review by pitrou — APPROVED** [review]
Everything looks fine to me. Just a nit in case you want to polish this PR.

File: python/pyarrow/tests/test_table.py (hunk 1 above)
Comment (tanyaschlusser) [reply python/pyarrow/tests/test_table.py]: Great idea! Incorporated in the next commit.

File: python/pyarrow/tests/test_table.py [hunk 2]
```diff
@@ -578,6 +578,14 @@ def test_table_basics():
 
     assert table.columns == columns
 
+    # Added to test https://issues.apache.org/jira/browse/ARROW-3866
```
[the remark this answers is not in the document]
Comment (tanyaschlusser) [reply python/pyarrow/tests/test_table.py]: Yes, that is better. Done!

**Comment by tanyaschlusser** [comment]
PS - Yes please @kszucs; this is about my level so I'd love to try similar issues.

**Pull request closed without merging**